framework/src/transfers/MultiAppNearestNodeTransfer.C

```diff
@@ -546,19 +538,22 @@ MultiAppNearestNodeTransfer::execute()
           // point. If there are multiple values from other processors
           // which are equidistant, the first one we check will "win".
           Real min_dist = std::numeric_limits<Real>::max();
-          for (unsigned int i_from = 0; i_from < incoming_evals.size(); i_from++)
+          for (auto & evals : incoming_evals)
           {
+            // processor Id
+            auto i_from = evals.first;
```

Comment:
Same here - you should switch this to being named `pid` or something of the sort.